libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -525,6 +527,50 @@ void AP_DDS_Client::update_topic(geographic_msgs_msg_GeoPointStamped& msg)
     }
 }
 
+bool AP_DDS_Client::update_topic(ardupilot_msgs_msg_Status& msg)
+{
+    // Fill the new message.
+    msg.vehicle_type = static_cast<uint8_t>(AP::fwversion().vehicle_type);
+    msg.armed = AP_Notify::flags.armed;
+    msg.mode = AP::vehicle()->get_mode();
```

Comment:
should do const auto &vehicle = AP::vehicle();

Reply:
set as suggested